src/main/java/net/sf/rails/ui/swing/ORUIManager.java

```diff
@@ -1428,9 +1428,13 @@ public void updateStatus(PossibleAction actionToComplete, boolean myTurn) {
 
             // TEMPORARY extra message about having no route
             for (BuyTrain bTrain : possibleActions.getType(BuyTrain.class)) {
-                if (bTrain.isForcedBuyIfNoRoute()) {
+                if (bTrain.isForcedBuyIfHasRoute()) {
```

Comment:
The performed check does not fit with the comment above. Is this change intentional (I do not really know the performed logic here just want to be sure so that we don't introduce an error)?

Reply:
The comment is old, I have not looked at it, other than recognizing that TEMPORARY means that automatic having-a-route checking has not yet been implemented.

Let's call the two conditions:
A = must have a train if the company has a route.
B = must have a train if the company has no route.
Now if B is true, A is always true as well, but if A is true, B can have both values.
So if A is true, a message is displayed, the contents of which depends on B. 
This is the intended, and I believe also actual, effect of this nested if construct.